src/Generator/SchemaMapperGenerator.php

```diff
@@ -390,22 +389,27 @@ protected function generateMapStatementsForObject(Field $root, Variable $payload
                     $optionalVar = $optionalResponseItems[$i];
                 }
 
-                if ($root->hasOneOf()) {
-                    $tryStatement = $this->builder->expr(
-                        $this->builder->methodCall(
-                            $schemaVar,
-                            $this->getSetMethodName($field),
-                            [$optionalVar]
-                        )
-                    );
+                if ($root->hasOneOf() || $root->hasAnyOf()) {
+                    $tryStatements = [
+                        $this->builder->expr(
+                            $this->builder->methodCall(
+                                $schemaVar,
+                                $this->getSetMethodName($field),
+                                [$optionalVar]
+                            )
+                        ),
+                    ];
+                    if ($root->hasOneOf()) {
+                        $tryStatements[] = $this->builder->return($schemaVar);
```

Comment:
I don't think this return is doing much, my suggestions with that $matches would already discard the need for this return when it's oneOf